anti-tracking-threat-model.md

```diff
@@ -0,0 +1,196 @@
+   the query, path, or even hostname.
+1. Instead of signing `news.example`'s content directly, AdTech embeds the
+   user's identity in that content and signs the result on the fly.
+1. This successfully transfers the unique ID to the `news.example` JS
+   environment, where it can be picked up by advertising code there.
+
+### Proposed Mitigations
+
+#### Preflight to publisher
+
+1. The server responding with the signed package is required to send the
+   signature up front. This does not prevent any attacks but increases the
+   user-visible latency, which AdTech experiences as a cost.
+2. The user agent makes an ephemeral, cookie-less preflight request to
+   `news.example` to get the signature and then validates the package from
+   `adtech.example` against that signature.
+
+   Both the
+   [fully-offline](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html#fully-offline-use)
+   use case and [cryptographic
+   agility](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html#crypto-agility)
+   require `news.example` to be able to declare that multiple signatures are
+   currently valid. This requires AdTech to send all of its signatures to
+   `news.example` and for News to list them all in the now-quite-large preflight
+   response. Having enough signatures that each one identifies a user would be
+   detectable by clients and might be enough cost that News wouldn't be willing
+   to do it.
+3. We add a signed time stamp to the package signature to avoid AdTech telling
+   News to get signatures from `adtech.example` backend and send personalized
+   signatures back as preflight responses. With such time stamps, the user agent
+   can decide to not accept signatures younger than, say one minute. For this to
+   work we need signed, official time.
+
+   TODO: Figure out how this blocks the attack.
+
+#### Public signature repository
+
+Another potential mitigation would be some kind of public repository of
+signatures to check against.
+
+This does not prevent any attacks, but could make them detectable.
+
+#### Make package loads stateless
+
+When requesting a signed package:
+
+1. The request to the distributor must be credential-less. i.e. the [credentials
+   mode](https://fetch.spec.whatwg.org/#concept-request-credentials-mode) must
+   be `"omit"`. This prevents AdTech from learning the user's identity from its
+   first-party cookies.
+1. It must be an HTTP GET request to prevent, for example, a POST request body
+   from including a user ID.
+1. The request URL on the distributor must not have a query string. Fragments
+   aren't sent to the server and would be blocked after the redirect, if
+   necessary, by anti-tracking measures that are independent of web packages.
+1. The path of the package request must be the same as the path on the target
+   domain to prevent the distributor from encoding a user ID in the path.
```

Comment:
It also seems like the distributor can use the referrer in order to include a UID (e.g. redirect through a unique path, and have a permissive ReferrerPolicy). Mitigation can be to enforce stricter ReferrerPolicy on redirects or in general.

Reply:
As I'm saying in https://github.com/WICG/webpackage/pull/424#discussion_r279031642, I think we'll need to block that communication route on all redirects, so it's not special to web packages. Do you want me to add a separate attack describing it below?